cactusbot/handlers/events.py

```diff
@@ -187,3 +120,26 @@ def __init__(self, cache_data, api):
                 "join": values["join"],
                 "leave": values["leave"]
             }
+
+    async def _cache(self, packet, event):
+        response = MessagePacket(
+            self.alert_messages[event]["message"].replace(
+                "%USER%", packet.user
+            ))
+
+        if packet.success:
+            if self.cache_data["cache_{}".format(event)]:
+                if hasattr(self, "cached_{}".format(event)):
+                    cached = getattr(self, "cached_{}".format(event))
```

Comment:
Using `hasattr` and `getattr` here is unnecessary after you've made the above change.

```python
if "cached_{}".format(event) in self.cached_events:
    cached = self.cached_events["cached_{}".format(event)]
```